polaris-react/src/components/Banner/components/PolarisSummerEditions2023Banner/PolarisSummerEditions2023Banner.tsx

```diff
@@ -0,0 +1,154 @@
+
+import {useBannerColors} from './utilities';
+
+export function PolarisSummerEditions2023Banner({
+  status = 'info',
+  icon,
+  hideIcon,
+  onDismiss,
+  action,
+  secondaryAction,
+  title,
+  children,
+}: BannerProps) {
+  const withinContentContainer = useContext(WithinContentContext);
+  const {smDown} = useBreakpoints();
+  const {iconRGBA, backgroundColor, textColor, statusIcon, closeIcon} =
+    useBannerColors(status);
+
+  const BannerIcon = () => {
+    if (hideIcon) return null;
+
+    return (
+      <Box paddingInlineStart="05">
+        {icon ? (
+          <span style={{fill: iconRGBA}}>
+            <Icon source={icon} />
+          </span>
+        ) : (
+          <Icon source={statusIcon} />
+        )}
+      </Box>
+    );
+  };
+
+  const DismissButton = () => {
+    if (!onDismiss) return null;
+
+    return (
+      <Button
+        plain
+        icon={closeIcon}
+        onClick={onDismiss}
+        accessibilityLabel="Dismiss notification"
+      />
+    );
+  };
+
+  const ActionButtons = () => {
+    if (!action && !secondaryAction) return null;
+
+    return (
+      <ButtonGroup>
+        {action && (
+          <Button onClick={action.onAction} {...action}>
+            {action.content}
+          </Button>
+        )}
+        {secondaryAction && (
+          <Button onClick={secondaryAction.onAction} {...secondaryAction}>
+            {secondaryAction.content}
+          </Button>
+        )}
+      </ButtonGroup>
+    );
+  };
+
+  const BannerTitle = () => {
+    if (!title) return null;
+
+    return (
+      <Text as="h2" variant="headingSm" breakWord>
+        {title}
+      </Text>
+    );
+  };
+
+  const hasContent = children || action || secondaryAction;
+
+  return withinContentContainer ? (
+    <Box
+      width="100%"
+      background={backgroundColor}
+      padding="2"
+      borderRadius="2"
+      color={textColor}
+    >
+      <HorizontalStack align="space-between" blockAlign="start" wrap={false}>
+        <Box paddingInlineEnd={onDismiss ? '6' : undefined}>
+          <HorizontalStack gap="2" wrap={false}>
+            <BannerIcon />
+            <VerticalStack gap="2">
+              <VerticalStack gap="05">
+                <BannerTitle />
+                {children}
+              </VerticalStack>
+              <ActionButtons />
+            </VerticalStack>
+          </HorizontalStack>
+        </Box>
+        <Box position="absolute" insetBlockStart="3" insetInlineEnd="3">
+          <DismissButton />
+        </Box>
+      </HorizontalStack>
+    </Box>
+  ) : (
+    <Box width="100%">
+      <VerticalStack align="space-between">
+        <Box
+          background={backgroundColor}
+          color={textColor}
+          borderRadiusStartStart={smDown ? undefined : '2'}
+          borderRadiusStartEnd={smDown ? undefined : '2'}
+          borderRadiusEndStart={hasContent || smDown ? undefined : '2'}
+          borderRadiusEndEnd={hasContent || smDown ? undefined : '2'}
```

Comment:
note for later, we should probably accept '0' as a value for `borderRadius`. right now the only way to get no border radius is `undefined` and can't be used in the breakpoint object